pallets/parachain-staking/src/migration.rs

```diff
@@ -145,10 +145,19 @@ pub fn enable_staking<T: Config>() -> Weight {
 pub struct EnableStaking<T>(PhantomData<T>);
 impl<T: Config> OnRuntimeUpgrade for EnableStaking<T> {
     fn on_runtime_upgrade() -> Weight {
-        let onchain_version = Pallet::<T>::on_chain_storage_version();
+        let current = Pallet::<T>::current_storage_version();
+        let onchain = Pallet::<T>::on_chain_storage_version();
 
-        if onchain_version < 1 {
+        log::info!(
+            "💽 Running migration with current storage version {:?} / onchain {:?}",
+            current,
+            onchain
+        );
+
+        if current == 1 && onchain == 0 {
```

Comment:
Do we need this? Isn't this covered by the existing logic? I think we only want to run the migration if the `onchain` version is less than 1 right? Do we care about the current version?

Reply:
Depends I suppose.
The migration should run when the current implementation of the pallet is version 1.
Once we have a version 2, we can make a judgement call if it should be run or not. Till then its safer keeping it under this constrain.
Substrate pallets have similar conditions on upgrade.

Reply:
I don't think so, unless I am missing something. Regardless of the current implementation, I think the migration should run when the **onchain** version is < 1.